src/arvgvinterface.c

```diff
@@ -353,8 +360,20 @@ struct _ArvGvInterfaceClass {
 
 G_DEFINE_TYPE_WITH_CODE (ArvGvInterface, arv_gv_interface, ARV_TYPE_INTERFACE, G_ADD_PRIVATE (ArvGvInterface))
 
+void
+arv_gv_interface_set_discovery_interface_name (ArvInterface *interface, const char *discovery_interface)
+{
+	ARV_GV_INTERFACE (interface)->priv->discovery_interface = strdup (discovery_interface);
```

Comment:
If discovery_interface was already set, a memory leak occurs. Free the content before setting the new one. Use g_strdup instead of strdup.